testing/functest_api_v20.py

```diff
@@ -558,6 +558,58 @@ def test_create_result_invalid_data(self):
         assert r.status_code == 400
         assert data['message'].startswith("Colon not allowed in key name:")
 
+    def test_create_result_submit_time_as_number(self):
+        ref_data = json.dumps(dict(
+            outcome=self.ref_result_outcome,
+            testcase=self.ref_testcase,
+            submit_time=1661324097123,
+        ))
+
+        r = self.app.post('/api/v2.0/results', data=ref_data, content_type='application/json')
+        data = json.loads(r.data)
+
+        assert r.status_code == 201
+        assert data['submit_time'] == '2022-08-24T06:54:57.123000'
+
+    def test_create_result_submit_time_as_number_string(self):
+        ref_data = json.dumps(dict(
+            outcome=self.ref_result_outcome,
+            testcase=self.ref_testcase,
+            submit_time="1661324097123",
+        ))
+
+        r = self.app.post('/api/v2.0/results', data=ref_data, content_type='application/json')
+        data = json.loads(r.data)
+
+        assert r.status_code == 201
+        assert data['submit_time'] == '2022-08-24T06:54:57.123000'
+
+    def test_create_result_submit_time_as_datetime(self):
+        ref_data = json.dumps(dict(
+            outcome=self.ref_result_outcome,
+            testcase=self.ref_testcase,
+            submit_time='2022-08-24T06:54:57.123456',
+        ))
+
+        r = self.app.post('/api/v2.0/results', data=ref_data, content_type='application/json')
+        data = json.loads(r.data)
+
+        assert r.status_code == 201
+        assert data['submit_time'] == '2022-08-24T06:54:57.123456'
+
+    def test_create_result_submit_time_as_invalid(self):
+        ref_data = json.dumps(dict(
+            outcome=self.ref_result_outcome,
+            testcase=self.ref_testcase,
+            submit_time='now',
+        ))
+
+        r = self.app.post('/api/v2.0/results', data=ref_data, content_type='application/json')
+        data = json.loads(r.data)
+
+        assert r.status_code == 400
+        assert data['message'].startswith('Malformed Request')
```

Comment:
What is the whole message? Is it containing "Expected timestamp in milliseconds or datetime...". I am thinking the message should be friendly for the user.

Reply:
It should contain the whole message. Unfortunately, there is a bug in flask-restful: https://github.com/flask-restful/flask-restful/issues/836

We should migrate the code since reqparse is deprecated. I will create an issue for this.

Reply:
Ok. Then that is fine for me.